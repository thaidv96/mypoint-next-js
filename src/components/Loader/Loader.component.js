import React from 'react'
import PropTypes from 'prop-types'
import styles from './Loader.styles'

const Loader = ({
  isFetching,
}) => {
  const loaderClassname = [
    'Loader',
    isFetching && 'loading',
  ]
    .filter(Boolean)
    .join(' ')

  return (
    <div
      className={loaderClassname}
    >
      <style jsx>{styles}</style>
    </div>
  )
}

Loader.propTypes = {
  isFetching: PropTypes.bool,
}

Loader.defaultProps = {
  isFetching: false,
}

export default Loader
