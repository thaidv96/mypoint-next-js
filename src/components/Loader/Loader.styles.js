import css from 'styled-jsx/css'

export default css`
  @keyframes rotate {
    from {
      transform: rotate(0);
    }
    to {
      transform: rotate(359.99deg);
    }
  }

  .Loader {
    position: absolute;
    top: 1em;
    right: 1em;
    opacity: 0;
    display: block;
    width: 2.5em;
    height: 2.5em;
    box-sizing: border-box;
    border-radius: 50%;
    border-width: 3px;
    border-style: solid;
    border-color: #444 transparent #444 transparent;
    animation: rotate 0.5s linear infinite;
    transition: opacity 0.2s;
  }

  .loading {
    opacity: 1;
    mix-blend-mode: color-burn;
  }
`
