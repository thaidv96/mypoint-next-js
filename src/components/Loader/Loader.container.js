import { connect } from 'react-redux'

import Loader from './Loader.component'

const mapStateToProps = ({
  network: {
    isFetching,
  },
}) => ({
  isFetching,
})

export default connect(mapStateToProps)(Loader)
