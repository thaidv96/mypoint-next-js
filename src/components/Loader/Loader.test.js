/* eslint-env jest */
/* global window */
import React from 'react'
import { shallow } from 'enzyme'

import Loader from './Loader.component'

describe('Loader', () => {
  const wrapperVisible = shallow(
    <Loader
      isFetching
    />
  )

  const wrapperNotVisible = shallow(
    <Loader />
  )

  it('matches its snapshots', () => {
    expect(wrapperVisible).toMatchSnapshot('visible')
    expect(wrapperNotVisible).toMatchSnapshot('notVisible')
  })
})
