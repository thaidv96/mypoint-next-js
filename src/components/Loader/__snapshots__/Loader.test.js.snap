// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader matches its snapshots: notVisible 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loader
    isFetching={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <JSXStyle
        css="@-webkit-keyframes rotate-jsx-4062756133{from{-webkit-transform:rotate(0);-ms-transform:rotate(0);transform:rotate(0);}to{-webkit-transform:rotate(359.99deg);-ms-transform:rotate(359.99deg);transform:rotate(359.99deg);}}@keyframes rotate-jsx-4062756133{from{-webkit-transform:rotate(0);-ms-transform:rotate(0);transform:rotate(0);}to{-webkit-transform:rotate(359.99deg);-ms-transform:rotate(359.99deg);transform:rotate(359.99deg);}}.Loader.jsx-4062756133{position:absolute;top:1em;right:1em;opacity:0;display:block;width:2.5em;height:2.5em;box-sizing:border-box;border-radius:50%;border-width:3px;border-style:solid;border-color:#444 transparent #444 transparent;-webkit-animation:rotate-jsx-4062756133 0.5s linear infinite;animation:rotate-jsx-4062756133 0.5s linear infinite;-webkit-transition:opacity 0.2s;transition:opacity 0.2s;}.loading.jsx-4062756133{opacity:1;mix-blend-mode:color-burn;}"
        styleId="4062756133"
      />,
      "className": "jsx-4062756133 Loader",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "css": "@-webkit-keyframes rotate-jsx-4062756133{from{-webkit-transform:rotate(0);-ms-transform:rotate(0);transform:rotate(0);}to{-webkit-transform:rotate(359.99deg);-ms-transform:rotate(359.99deg);transform:rotate(359.99deg);}}@keyframes rotate-jsx-4062756133{from{-webkit-transform:rotate(0);-ms-transform:rotate(0);transform:rotate(0);}to{-webkit-transform:rotate(359.99deg);-ms-transform:rotate(359.99deg);transform:rotate(359.99deg);}}.Loader.jsx-4062756133{position:absolute;top:1em;right:1em;opacity:0;display:block;width:2.5em;height:2.5em;box-sizing:border-box;border-radius:50%;border-width:3px;border-style:solid;border-color:#444 transparent #444 transparent;-webkit-animation:rotate-jsx-4062756133 0.5s linear infinite;animation:rotate-jsx-4062756133 0.5s linear infinite;-webkit-transition:opacity 0.2s;transition:opacity 0.2s;}.loading.jsx-4062756133{opacity:1;mix-blend-mode:color-burn;}",
        "styleId": "4062756133",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <JSXStyle
          css="@-webkit-keyframes rotate-jsx-4062756133{from{-webkit-transform:rotate(0);-ms-transform:rotate(0);transform:rotate(0);}to{-webkit-transform:rotate(359.99deg);-ms-transform:rotate(359.99deg);transform:rotate(359.99deg);}}@keyframes rotate-jsx-4062756133{from{-webkit-transform:rotate(0);-ms-transform:rotate(0);transform:rotate(0);}to{-webkit-transform:rotate(359.99deg);-ms-transform:rotate(359.99deg);transform:rotate(359.99deg);}}.Loader.jsx-4062756133{position:absolute;top:1em;right:1em;opacity:0;display:block;width:2.5em;height:2.5em;box-sizing:border-box;border-radius:50%;border-width:3px;border-style:solid;border-color:#444 transparent #444 transparent;-webkit-animation:rotate-jsx-4062756133 0.5s linear infinite;animation:rotate-jsx-4062756133 0.5s linear infinite;-webkit-transition:opacity 0.2s;transition:opacity 0.2s;}.loading.jsx-4062756133{opacity:1;mix-blend-mode:color-burn;}"
          styleId="4062756133"
        />,
        "className": "jsx-4062756133 Loader",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": "@-webkit-keyframes rotate-jsx-4062756133{from{-webkit-transform:rotate(0);-ms-transform:rotate(0);transform:rotate(0);}to{-webkit-transform:rotate(359.99deg);-ms-transform:rotate(359.99deg);transform:rotate(359.99deg);}}@keyframes rotate-jsx-4062756133{from{-webkit-transform:rotate(0);-ms-transform:rotate(0);transform:rotate(0);}to{-webkit-transform:rotate(359.99deg);-ms-transform:rotate(359.99deg);transform:rotate(359.99deg);}}.Loader.jsx-4062756133{position:absolute;top:1em;right:1em;opacity:0;display:block;width:2.5em;height:2.5em;box-sizing:border-box;border-radius:50%;border-width:3px;border-style:solid;border-color:#444 transparent #444 transparent;-webkit-animation:rotate-jsx-4062756133 0.5s linear infinite;animation:rotate-jsx-4062756133 0.5s linear infinite;-webkit-transition:opacity 0.2s;transition:opacity 0.2s;}.loading.jsx-4062756133{opacity:1;mix-blend-mode:color-burn;}",
          "styleId": "4062756133",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Loader matches its snapshots: visible 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loader
    isFetching={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <JSXStyle
        css="@-webkit-keyframes rotate-jsx-4062756133{from{-webkit-transform:rotate(0);-ms-transform:rotate(0);transform:rotate(0);}to{-webkit-transform:rotate(359.99deg);-ms-transform:rotate(359.99deg);transform:rotate(359.99deg);}}@keyframes rotate-jsx-4062756133{from{-webkit-transform:rotate(0);-ms-transform:rotate(0);transform:rotate(0);}to{-webkit-transform:rotate(359.99deg);-ms-transform:rotate(359.99deg);transform:rotate(359.99deg);}}.Loader.jsx-4062756133{position:absolute;top:1em;right:1em;opacity:0;display:block;width:2.5em;height:2.5em;box-sizing:border-box;border-radius:50%;border-width:3px;border-style:solid;border-color:#444 transparent #444 transparent;-webkit-animation:rotate-jsx-4062756133 0.5s linear infinite;animation:rotate-jsx-4062756133 0.5s linear infinite;-webkit-transition:opacity 0.2s;transition:opacity 0.2s;}.loading.jsx-4062756133{opacity:1;mix-blend-mode:color-burn;}"
        styleId="4062756133"
      />,
      "className": "jsx-4062756133 Loader loading",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "css": "@-webkit-keyframes rotate-jsx-4062756133{from{-webkit-transform:rotate(0);-ms-transform:rotate(0);transform:rotate(0);}to{-webkit-transform:rotate(359.99deg);-ms-transform:rotate(359.99deg);transform:rotate(359.99deg);}}@keyframes rotate-jsx-4062756133{from{-webkit-transform:rotate(0);-ms-transform:rotate(0);transform:rotate(0);}to{-webkit-transform:rotate(359.99deg);-ms-transform:rotate(359.99deg);transform:rotate(359.99deg);}}.Loader.jsx-4062756133{position:absolute;top:1em;right:1em;opacity:0;display:block;width:2.5em;height:2.5em;box-sizing:border-box;border-radius:50%;border-width:3px;border-style:solid;border-color:#444 transparent #444 transparent;-webkit-animation:rotate-jsx-4062756133 0.5s linear infinite;animation:rotate-jsx-4062756133 0.5s linear infinite;-webkit-transition:opacity 0.2s;transition:opacity 0.2s;}.loading.jsx-4062756133{opacity:1;mix-blend-mode:color-burn;}",
        "styleId": "4062756133",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <JSXStyle
          css="@-webkit-keyframes rotate-jsx-4062756133{from{-webkit-transform:rotate(0);-ms-transform:rotate(0);transform:rotate(0);}to{-webkit-transform:rotate(359.99deg);-ms-transform:rotate(359.99deg);transform:rotate(359.99deg);}}@keyframes rotate-jsx-4062756133{from{-webkit-transform:rotate(0);-ms-transform:rotate(0);transform:rotate(0);}to{-webkit-transform:rotate(359.99deg);-ms-transform:rotate(359.99deg);transform:rotate(359.99deg);}}.Loader.jsx-4062756133{position:absolute;top:1em;right:1em;opacity:0;display:block;width:2.5em;height:2.5em;box-sizing:border-box;border-radius:50%;border-width:3px;border-style:solid;border-color:#444 transparent #444 transparent;-webkit-animation:rotate-jsx-4062756133 0.5s linear infinite;animation:rotate-jsx-4062756133 0.5s linear infinite;-webkit-transition:opacity 0.2s;transition:opacity 0.2s;}.loading.jsx-4062756133{opacity:1;mix-blend-mode:color-burn;}"
          styleId="4062756133"
        />,
        "className": "jsx-4062756133 Loader loading",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": "@-webkit-keyframes rotate-jsx-4062756133{from{-webkit-transform:rotate(0);-ms-transform:rotate(0);transform:rotate(0);}to{-webkit-transform:rotate(359.99deg);-ms-transform:rotate(359.99deg);transform:rotate(359.99deg);}}@keyframes rotate-jsx-4062756133{from{-webkit-transform:rotate(0);-ms-transform:rotate(0);transform:rotate(0);}to{-webkit-transform:rotate(359.99deg);-ms-transform:rotate(359.99deg);transform:rotate(359.99deg);}}.Loader.jsx-4062756133{position:absolute;top:1em;right:1em;opacity:0;display:block;width:2.5em;height:2.5em;box-sizing:border-box;border-radius:50%;border-width:3px;border-style:solid;border-color:#444 transparent #444 transparent;-webkit-animation:rotate-jsx-4062756133 0.5s linear infinite;animation:rotate-jsx-4062756133 0.5s linear infinite;-webkit-transition:opacity 0.2s;transition:opacity 0.2s;}.loading.jsx-4062756133{opacity:1;mix-blend-mode:color-burn;}",
          "styleId": "4062756133",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
